packages/react-testing/src/mount.ts

```diff
@@ -105,11 +105,11 @@ export function createMount<
     options: MountOptions = {} as any,
   ) {
     const context = createContext(options);
-    const rendered = render(element, context, options);
 
-    const wrapper = new CustomRoot(rendered, context, root =>
-      root.find(element.type),
-    );
+    const wrapper = new CustomRoot(element, context, {
+      render: element => render(element, context, options),
+      resolveRoot: root => root.find(element.type),
+    });
```

Comment:
The basic gist of this fix is that instead of a `Root` around the result of `render(element)`, the `Root` just gets the raw element under test, and we thread through the `render` to the component that eventually renders the component to the DOM, since it is in charge of handling updated props.